src/modm/driver/storage/cat24aa.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020, Mike Wolfram
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_CAT24AA_HPP
+#define MODM_CAT24AA_HPP
+
+#include <modm/processing/resumable.hpp>
+#include <modm/architecture/interface/i2c_device.hpp>
+
+namespace modm
+{
+
+/// @cond
+struct cat24Aa {
+	class DataTransmissionAdapter : public modm::I2cWriteReadTransaction
+	{
+	public:
+		DataTransmissionAdapter(uint8_t address = 0x50);
+
+		bool
+		configureWrite(uint16_t address, const uint8_t *buffer, std::size_t size);
+
+		bool
+		configureRead(uint16_t address, uint8_t *buffer, std::size_t size);
+
+		inline static constexpr uint8_t getAddress() {
+			return 0x50;
+		}
+
+	protected:
+		virtual Writing
+		writing() override;
+
+		uint8_t addressBuffer[1];
+		bool writeAddress;
+	};
+};
+/// @endcond
+
+/**
+ * I2C Eeprom CAT24AA
+ *
+ * Driver for OnSemi CAT24AA Eeprom.
+ * I2C eeprom with an 16-bit address pointer, encoded in 3 bits of the device
+ * slave address plus the address byte.
+ * Base address is fixed to 0x50.
+ *
+ * @ingroup	modm_driver_i2c_eeprom
+ * @author	Fabian Greif
+ * @author	Niklas Hauser
```

Comment:
Update with your name, this is mostly for knowing who to ask for help/bugs/features.

Reply:
I should stop working in the hotel bar and start wearing my glasses. :-)